editor/src/components/canvas/canvas-strategies/strategies/reparent-strategy-helpers.ts

```diff
@@ -987,6 +1022,51 @@ export function getReparentPropertyChanges(
       )
     case 'REPARENT_AS_STATIC':
       const newPath = EP.appendToPath(newParent, EP.toUid(target))
-      return getStaticReparentPropertyChanges(newPath, targetOriginalStylePosition)
+      const directions = getDirectionsForValidStaticReparentTarget(
+        newParent,
+        newParentStartingMetadata,
+      )
+
+      const convertDisplayInline =
+        directions === 'non-1d-static' ||
+        directions.direction == null ||
+        directions.flexOrFlow === 'flex'
+          ? 'do-not-convert'
+          : directions.direction
+
+      return getStaticReparentPropertyChanges(
+        newPath,
+        targetOriginalStylePosition,
+        targetOriginalDisplayProp,
+        convertDisplayInline,
+      )
+  }
+}
+
+function getDirectionsForValidStaticReparentTarget(
+  path: ElementPath,
+  metadata: ElementInstanceMetadataMap,
+):
+  | {
```

Comment:
Is this the same return type that `staticContainerDirections` has?

Reply:
fixed!